src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -517,7 +518,16 @@ private String setupGlobalSettingFile() throws IOException, InterruptedException
      * @throws AbortException in case of error
      */
     private void settingsFromConfig(String settingsConfigId, FilePath settingsFile) throws AbortException {
-        Config c = Config.getByIdOrNull(settingsConfigId);
+
+        Config c = null;
+        Executor executor = Executor.currentExecutor();
+        if (executor != null) {
+            Queue.Executable currentExecutable = executor.getCurrentExecutable();
+            if (currentExecutable != null) {
+                c = ConfigFiles.getByIdOrNull((Run<?, ?>) currentExecutable, settingsConfigId);

```

Comment:
I just checked, this unfortunate, this does not work @jglick do you have any hint how I can access the Run/Build from a `StepContext`? I need to to be able to identify the context where to look for config-files (e.g. in witch folder)